ambry-server/src/main/java/com.github.ambry.server/AmbryServerRequests.java

```diff
@@ -65,21 +67,29 @@
  * handled by this class
  */
 public class AmbryServerRequests extends AmbryRequests {
-
-  private final ConcurrentHashMap<PartitionId, ReplicaId> localPartitionToReplicaMap;
+  private final ServerConfig serverConfig;
   private final StatsManager statsManager;
   private final ConcurrentHashMap<RequestOrResponseType, Set<PartitionId>> requestsDisableInfo =
       new ConcurrentHashMap<>();
+  private final ConcurrentHashMap<PartitionId, ReplicaId> localPartitionToReplicaMap;
+  // POST requests are allowed on stores states: { LEADER, STANDBY }
+  static final Set<ReplicaState> POST_ALLOWED_STORE_STATES = EnumSet.of(ReplicaState.LEADER, ReplicaState.STANDBY);
+  // UPDATE requests (including DELETE, TTLUpdate) are allowed on stores states: { LEADER, STANDBY, INACTIVE, BOOTSTRAP }
+  static final Set<ReplicaState> UPDATE_ALLOWED_STORE_STATES =
+      EnumSet.of(ReplicaState.LEADER, ReplicaState.STANDBY, ReplicaState.INACTIVE, ReplicaState.BOOTSTRAP);
+  static final Set<RequestOrResponseType> UPDATE_REQUEST_TYPES =
+      EnumSet.of(RequestOrResponseType.DeleteRequest, RequestOrResponseType.TtlUpdateRequest);
```

Comment:
we have UNDELETE now.

Reply:
I will temporarily defer the UNDELETE until the logic is fully implemented.